### DISCUSSION TITLE
Cannot insert

### DISCUSSION THREAD

**Discussion by phuoccss** [supabase/supabase#1150] [Questions] [answered]
Hi I have a table with following schema.

```sql
CREATE TABLE public.projects (
    id uuid PRIMARY KEY DEFAULT uuid_generate_v4 (),
    creator_id uuid NOT NULL REFERENCES auth.users (id),
    ...
);
```

I enabled RLS like below.
```sql
CREATE POLICY "Allow individual insert access" on public.projects FOR INSERT WITH CHECK ( auth.uid() = creator_id );
```
![Screen Shot 2021-04-15 at 17 59 43](https://user-images.githubusercontent.com/15166715/114692831-91d9a500-9d4b-11eb-8ed3-9fea52570759.png)

I always get this error.
![image](https://user-images.githubusercontent.com/15166715/114693256-001e6780-9d4c-11eb-860c-46d99f40e1da.png)

I even tried ... WITH CHECK (true) and still got the same error. It's incredibly frustrating.

**Comment by inian** [comment, marked as answer]
Can you add returning minimal to your query? By default postgrest selects and returns the object as part of the insert query. You can disable this behaviour by doing this - 
```
await supabase.from('projects').insert([{ creator_id: 123 }], { returning: 'minimal' })
```

Or add a select policy for the user too and not change your JS code.

**Reply by phuoccss** [reply]
Got it. Thanks for the fast response.

**Reply by PedroLoboA** [reply]
thanks! i was strugling with that problem to.

**Reply by PedroLoboA** [reply]
one question, 

if i do  { returning: 'minimal' } i am not able to subscribe(), and if i dont put { returning: 'minimal' } i get a error.

this is happening because i have a foreign key in the table, i would need to insert into two tables before get the return, otherwise i will get the error. thats some way to do that?
